autoload/lsp/util.vim

```diff
@@ -112,8 +112,10 @@ export def LspFileToUri(fname: string): string
   var uri: string = fname->fnamemodify(':p')
 
   if has("win32unix")
-    # We're in Cygwin
-    uri = CygwinToWindowsPath(uri)
+    # We're in Cygwin, we need to convert POSIX style paths to Windows style.
+    # `cygpath -m` converts paths of the form "/cygdrive/c/foo/bar" to "C:/foo/bar", and
+    # paths of the form "/home/pete/foo" to "C:/cygwin64/home/pete/foo".
+    uri = system($'cygpath -m {uri}')->substitute('^\(\p*\).*$', '\=submatch(1)', "")
```

Comment:
this substitute seems not very efficient or necessary?

Reply:
The substitution is to remove end of line escape characters. Quick internet searching did not show me a better alternative.

Reply:
i think this perhaps can be refined a bit, but maybe not a matter.

Reply:
Can you suggest how? As I mentioned in the other PR, I'm not a seasoned vim scripter and google search did not provide better alternatives.

Reply:
let it be, not a matter.

Reply:
let it be, not a matter.